app/api/db/__init__.py

```diff
@@ -1,60 +1,159 @@
+"""
+Database module.
+
+It is recommended to import this module as `import api.db` rather than
+`from api import db` in order to avoid confusion with instances of `DB`
+which are named `db` by convention.
+
+Example:
+    import api.db
+
+    db = api.db.init_db()
+
+    # non-ORM style usage
+    with db.get_connection() as conn:
+        conn.execute(...)
+
+    # ORM style usage
+    with db.get_session() as session:
+        session.query(...)
+        with session.begin():
+            session.add(...)
+
+Alternatively, if you are only importing the module for type hints (and
+to explicitly indicate dependencies), you can do `from api.db import DB`.
+
+Example:
+    from api.db import DB
+
+    @app.route("/health")
+    def health() -> Response:
+        db: DB = app.extensions["db"]
+        with db.get_connection() as conn:
+            conn.execute(...)
+"""
+
 import os
 import urllib.parse
-from contextlib import contextmanager
-from typing import Any, Generator, Optional
+from typing import Any, Optional
 
 import psycopg2
+import sqlalchemy
 import sqlalchemy.pool as pool
-from sqlalchemy import create_engine
-from sqlalchemy.engine import Engine
-from sqlalchemy.orm import scoped_session, sessionmaker
+from flask import Flask
+from sqlalchemy.orm import session
 
 import api.logging
 from api.db.db_config import DbConfig, get_db_config
-from api.db.migrations.run import have_all_migrations_run
+
+FLASK_EXTENSION_KEY = "db"
```

Comment:
Maybe `FLASK_DB_EXTENSION_KEY` to distinguish it from any other one?

Reply:
i was thinking that this is a constant private to this file e.g. it's the FLASK_EXTENSION_KEY for the api.db module. what if instead i prefixed the constant with _ like `_FLASK_EXTENSION_KEY`, would that make it clearer?